changelog/unreleased/issue-2591

```diff
@@ -0,0 +1,22 @@
+Bugfix: Don't read password from stdin for backup --stdin
+
+Restic backup previously tried to read first the password, then the data
+to be backed up from standard input. This meant it would often confuse part
+of the data for the password.
+
+From now on, restic backup --stdin will exit with the message
+
+    Fatal: cannot read both password and data from stdin
+
+unless the password is passed in some other way (--restic-password-file,
+RESTIC_PASSWORD, etc.). To enter the password interactively, a password
+command has to be used. For example, on Linux,
+
+    mysqldump somedatabase |
+        restic backup --stdin \
+            --password-command='sh -c "systemd-ask-password < /dev/tty"'
```

Comment:
I must confess I'm not completely sure why the redirect is needed, but just

    --password-command=systemd-ask-password

doesn't work.

Reply:
`systemd-ask-password < /dev/null` for me also doesn't work. So I guess that is caused by the following issue: https://github.com/systemd/systemd/issues/1232